### PR TITLE
Deprecate method that throws deprecation along the way

### PR DESCRIPTION
**Pull request by dereuromark**
Follow https://github.com/cakephp/cakephp/pull/14777

### CODE REVIEW

File: src/View/Helper/NumberHelper.php [hunk 1]
```diff
@@ -222,6 +222,7 @@ public function formatDelta($value, array $options = []): string
      * if $currency argument is not provided. If boolean false is passed, it will clear the
      * currently stored value. Null reads the current default.
      * @return string|null Currency
+     * @deprecated 4.1.0 Use setDefaultCurrency()/getDefaultCurrency() instead.
```
Comment (saeideng) [inline src/View/Helper/NumberHelper.php:225]:
```suggestion
     * @deprecated 3.9.0 Use setDefaultCurrency()/getDefaultCurrency() instead.
```

Comment (saeideng) [reply src/View/Helper/NumberHelper.php:225]: if you check these methods exist in `4.0.0` too

Comment (dereuromark) [reply src/View/Helper/NumberHelper.php:225]: Yes but we just started to declare, no?

Comment (saeideng) [reply src/View/Helper/NumberHelper.php:225]: no , since 4.0.0
see https://github.com/cakephp/cakephp/blob/4.0.0/src/I18n/Number.php#L275-L278

**Pull request merged**